feat: persons population script

JuliusKryger opened the pull request:
Added a population script for all of the persons data, this will need to be ran after the core/media one. 

Also raised the value on the create scripts for person name and roles for crew and cast memebers. to 150 instead of 50.

Commit:
persons population script done

JuliusKryger committed:
Merge branch 'main' into feature/import-data-related-to-persons

sofushn commented on src/B2_build_movie_db.sql:
```diff
@@ -133,7 +133,7 @@ CREATE TABLE promotional_media (
 CREATE TABLE person (
     person_id       INTEGER     PRIMARY KEY GENERATED ALWAYS AS IDENTITY,
-    "name"          VARCHAR(50) NOT NULL,
+    "name"          VARCHAR(150) NOT NULL,
```
fix pls... (whitespace dose not match other rows)

sofushn commented on src/B2_build_movie_db.sql:
```diff
@@ -143,16 +143,16 @@ CREATE TABLE person (
 CREATE TABLE crew_member (
     crew_member_id  INTEGER     PRIMARY KEY GENERATED ALWAYS AS IDENTITY,
-    "role"          VARCHAR(50) NULL,
+    "role"          VARCHAR(150) NULL,
```
fix pls... (whitespace dose not match other rows)

sofushn commented on src/populate_persons.sql:
```diff
@@ -0,0 +1,45 @@
+
+INSERT INTO person (name, birth_date, death_date, "description", imdb_id)
+SELECT 
+    LEFT(primaryname, 50) AS name,  -- Truncate name to 50 characters to prevent errors
```
You change the type to allow for 150 chars, is this still needed?

sofushn commented on src/populate_persons.sql:
```diff
@@ -0,0 +1,45 @@
+    LEFT(primaryname, 50) AS name,  -- Truncate name to 50 characters to prevent errors
+    CASE WHEN birthyear = '' THEN NULL ELSE TO_DATE(birthyear, 'YYYY') END AS birth_date,
+    CASE WHEN deathyear = '' THEN NULL ELSE TO_DATE(deathyear, 'YYYY') END AS death_date,
+    primaryprofession AS description,
```
This seems wrong, I would just not set description

JuliusKryger replied:
Each person is associated with a list of roles like (actor, director, producer) from the IMDB_DATA. I just thought it would go in the description, if that is not the case, this can be left out for now.

JuliusKryger replied:
It has been changed to set the description as NULL now when inserting into persons.

sofushn commented on src/populate_persons.sql:
```diff
@@ -0,0 +1,45 @@
+    nconst AS imdb_id
+FROM original.name_basics;
+
+WITH job_roles AS (
+    SELECT DISTINCT category
+    FROM original.title_principals
+)
+INSERT INTO job_category (name)
+SELECT category
+FROM job_roles;
```
Is the with statement needed here? can't you just do the following?

```suggestion
INSERT INTO job_category (name)
SELECT DISTINCT category 
FROM original.title_principals
```

JuliusKryger replied:
No it is not.

sofushn commented on src/populate_persons.sql:
```diff
@@ -0,0 +1,45 @@
+JOIN media m ON t.tconst = m.imdb_id
+JOIN person p ON p.imdb_id = t.nconst
+JOIN job_category jc ON jc.name = t.category
+WHERE t.category IN ('director', 'producer', 'writer');
```
the `crew_member` should be everybody but `actor` and `actress`

Commit:
refactor: fixed whitepace issue

Commit:
refactor: fixed population script based on PR feedback

JuliusKryger commented:
**Summary of Changes:**

1. **Crew Members (Excluding Actors/Actresses):**
   - Updated the logic to insert all crew members into the `crew_member` table, while excluding entries where the `category` is either `'actor'` or `'actress'`. 
   
2. **Cast Members (Only Actors/Actresses):**
   - Ensured that only actors and actresses are inserted into the `cast_member` table by filtering categories strictly for `'actor'` and `'actress'`.

3. **Job Category Insertion:**
   - Removed the unnecessary `WITH` clause and directly inserted distinct job categories into the `job_category` table from `original.title_principals`.

4. **Person Table Insertion:**
   - Ensured that the `person` table is populated correctly with full names (no truncation), and handled potential `NULL` values for birth and death dates.
